File: CS/ExportReportDemo/ExportReportDemo.Module/DatabaseUpdate/Updater.cs
using System;
using System.Linq;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Persistent.BaseImpl;
using ExportReportDemo.Module.BusinessObjects;

namespace ExportReportDemo.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
            base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            CreateEmployees();
        }
        private void CreateEmployees()
        {
            AddEmployee("Andrew", "Fuller", "Sales Manager");
            AddEmployee("Mary", "Telitson", "Developer");
            AddEmployee("Michael", "Suyama", "Developer");
            AddEmployee("Nancy", "Davolio", "Developer");
            AddEmployee("Robert", "King", "Senior Developer");
            AddEmployee("Steven", "Buchanan", "Developer");
        }
        private void AddEmployee(string name, string lastName, string position)
        {
            Employee employee = ObjectSpace.FindObject<Employee>(CriteriaOperator.Parse(string.Format("FirstName == '{0}' && LastName == '{1}'", name, lastName)));
            if (employee == null)
            {
                employee = ObjectSpace.CreateObject<Employee>();
                employee.FirstName = name;
                employee.LastName = lastName;
                employee.Position = position;
            }
        }

    }
}

File: VB/ExportReportDemo/ExportReportDemo.Module/DatabaseUpdate/Updater.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.Updating
Imports ExportReportDemo.Module.BusinessObjects

Namespace ExportReportDemo.Module.DatabaseUpdate

    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub

        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            CreateEmployees()
        End Sub

        Private Sub CreateEmployees()
            AddEmployee("Andrew", "Fuller", "Sales Manager")
            AddEmployee("Mary", "Telitson", "Developer")
            AddEmployee("Michael", "Suyama", "Developer")
            AddEmployee("Nancy", "Davolio", "Developer")
            AddEmployee("Robert", "King", "Senior Developer")
            AddEmployee("Steven", "Buchanan", "Developer")
        End Sub

        Private Sub AddEmployee(ByVal name As String, ByVal lastName As String, ByVal position As String)
            Dim employee As Employee = ObjectSpace.FindObject(Of Employee)(CriteriaOperator.Parse(String.Format("FirstName == '{0}' && LastName == '{1}'", name, lastName)))
            If employee Is Nothing Then
                employee = ObjectSpace.CreateObject(Of Employee)()
                employee.FirstName = name
                employee.LastName = lastName
                employee.Position = position
            End If
        End Sub
    End Class
End Namespace
